src/hooks/useRovingCellRef.ts

```diff
@@ -2,32 +2,41 @@ import { useCallback, useState } from 'react';
 import { scrollIntoView } from '../utils';
 
 // https://www.w3.org/TR/wai-aria-practices-1.1/#kbd_roving_tabindex
-export function useRovingCellRef(isSelected: boolean) {
+export function useRovingCellRef(isSelected: boolean, isModeChangedFromEditToSelect?: boolean) {
   // https://www.w3.org/TR/wai-aria-practices-1.1/#gridNav_focus
   const [isChildFocused, setIsChildFocused] = useState(false);
 
   if (isChildFocused && !isSelected) {
     setIsChildFocused(false);
   }
 
-  const ref = useCallback((cell: HTMLDivElement | null) => {
-    if (cell === null) return;
-    scrollIntoView(cell);
-    if (cell.contains(document.activeElement)) return;
-    cell.focus({ preventScroll: true });
-  }, []);
+  const ref = useCallback(
+    (cell: HTMLDivElement | null) => {
+      if (cell === null) return;
+      scrollIntoView(cell);
+      if (
+        cell.contains(document.activeElement) ||
+        (isModeChangedFromEditToSelect && !cell.closest('.rdg')!.contains(document.activeElement))
```

Comment:
The idea is to check if the active element is outside the grid when the editor is closed. AFAICT `isModeChangedFromEditToSelect` is not needed and grid does not steal focus but some tests fail without it and I am not sure why. It also seems safer to keep it.